google-cloud-storage/src/main/java/com/google/cloud/storage/NotificationInfo.java

```diff
@@ -0,0 +1,297 @@
+      new Function<Notification, NotificationInfo>() {
+        @Override
+        public NotificationInfo apply(Notification pb) {
+          return NotificationInfo.fromPb(pb);
+        }
+      };
+  static final Function<NotificationInfo, Notification> TO_PB_FUNCTION =
+      new Function<NotificationInfo, Notification>() {
+        @Override
+        public Notification apply(NotificationInfo NotificationInfo) {
+          return NotificationInfo.toPb();
+        }
+      };
+  private final String generatedId;
+  private final ProjectTopicName topic;
+  private final List<String> eventTypes;
+  private final Map<String, String> customAttributes;
+  private final PayloadFormat payloadFormat;
+  private final String objectNamePrefix;
+  private final String etag;
+  private final String selfLink;
+
+  public static final class Builder {
+
+    private String generatedId;
+    private ProjectTopicName topic;
+    private List<String> eventTypes;
+    private Map<String, String> customAttributes;
+    private PayloadFormat payloadFormat;
+    private String objectNamePrefix;
+    private String etag;
+    private String selfLink;
+
+    Builder(ProjectTopicName topic) {
+      this.topic = topic;
+    }
+
+    Builder(NotificationInfo NotificationInfo) {
+      generatedId = NotificationInfo.generatedId;
+      etag = NotificationInfo.etag;
+      selfLink = NotificationInfo.selfLink;
+      topic = NotificationInfo.topic;
+      eventTypes = NotificationInfo.eventTypes;
+      customAttributes = NotificationInfo.customAttributes;
+      payloadFormat = NotificationInfo.payloadFormat;
+      objectNamePrefix = NotificationInfo.objectNamePrefix;
+    }
+
+    Builder setGeneratedId(String generatedId) {
+      this.generatedId = generatedId;
+      return this;
+    }
+
+    Builder setSelfLink(String selfLink) {
+      this.selfLink = selfLink;
+      return this;
+    }
+
+    public Builder setTopic(ProjectTopicName topic) {
+      this.topic = topic;
+      return this;
+    }
+
+    public Builder setPayloadFormat(PayloadFormat payloadFormat) {
+      this.payloadFormat = payloadFormat;
+      return this;
+    }
+
+    /** GcpLaunchStage.Alpha */
+    public Builder setObjectNamePrefix(String objectNamePrefix) {
+      this.objectNamePrefix = objectNamePrefix;
+      return this;
+    }
+
+    public Builder setEventTypes(Iterable<String> eventTypes) {
+      this.eventTypes = eventTypes != null ? ImmutableList.copyOf(eventTypes) : null;
+      return this;
+    }
+
+    Builder setEtag(String etag) {
+      this.etag = etag;
+      return this;
+    }
+
+    public Builder setCustomAttributes(Map<String, String> customAttributes) {
+      this.customAttributes =
+          customAttributes != null ? ImmutableMap.copyOf(customAttributes) : null;
+      return this;
+    }
+
+    public NotificationInfo build() {
+      checkNotNull(topic);
+      return new NotificationInfo(this);
+    }
+  }
+
+  NotificationInfo(Builder builder) {
```

Comment:
why can't we have this constructor as a `private`?

Reply:
done

Reply:
I still dont see this as a private or any reason for this to be package-private?